### DISCUSSION TITLE
Is there a way to automatically index the nodes?

### DISCUSSION THREAD

**Discussion by ooker777** [SkepticMystic/breadcrumbs#311] [Q&A]
Let's say I pick a node as the source/root node. It's children will be indexed as 1, 2, 3. Children of 1 will be indexed as 1.1, 1.2, 1.3, etc. Children of 1.1 will be indexed as 1.1.1, 1.1.2, 1.1.3, etc. Is there a way to automatically do that? Perhaps with the combination of the plugin Templater.

If I make the indexing more complicate, by identify that the children of 1 are 1.1a, 1.1b, 1.2, and occasionally there will be an "upgraded" version of a node, and it will be indexed by a plus, i.e. 1+, 1.1a+. I understand that these things need to be done manually, but is it possible to make the script assume that the indexing for the children of a node should be normal, before I manually edit it?

Related question: [Is there a way to make "close siblings"?](https://github.com/SkepticMystic/breadcrumbs/discussions/309)

**Comment by michaelpporter** [comment]
There's a related feature, but it works the other way around, so it's worth being clear about the direction.

Breadcrumbs has a **Johnny.Decimal** edge builder, but it's **input-only**: you name notes with the numeric prefix (e.g. `01.02 Title`) and Breadcrumbs *infers* the up/down hierarchy from those numbers. It reads numbering → hierarchy; it does **not** auto-assign or renumber (hierarchy → numbering).

So the specific ask here — automatically generating `1`, `1.1`, `1.1.1`, … for an existing parent/child tree — isn't something Breadcrumbs does today. If you're willing to write the numbers yourself, the Johnny.Decimal builder will turn them into a navigable hierarchy. To go the other way (emit the numbers from the graph), you'd combine Breadcrumbs' graph — via `window.BCAPI.create_list_index()` or `get_neighbours()` ([API docs](https://breadcrumbs-docs.michaelpporter.com/api/)) — with something like Templater or DataviewJS to write the index.

I'll leave this open as a feature request for **auto-numbering an existing hierarchy**, since that's genuinely not implemented.

(Your tail question about "close siblings" is tracked separately in #309.) 🍞

**Comment by michaelpporter** [comment]
Breadcrumbs reads numbering to build hierarchy (the Johnny.Decimal builder: name a note `01.02 Title` and it infers the up/down edges), but it deliberately doesn't go the other way — auto-stamping `1`, `1.1`, `1.1.1` onto an existing parent/child tree.

That direction is an authoring/automation task rather than an edge-display feature, so it sits outside the plugin's core. Two ways to get it today:

- **Ordered index:** the "Create list index" command walks down from a root note in order and outputs a nested list — the indentation already encodes the same hierarchy the numbers would.
- **True auto-numbering:** use `window.BCAPI` from Templater or DataviewJS to traverse the tree and write numbers into frontmatter/filenames however you like.

Keeping the Johnny.Decimal flow one-directional (numbers → hierarchy) is intentional, so I'm closing this as resolved. Thanks for the suggestion!
